src/stirling/source_connectors/socket_tracer/bcc_bpf/go_trace_common.h

```diff
@@ -174,13 +174,25 @@ static __inline int32_t get_fd_from_conn_intf_core(struct go_interface conn_intf
   REQUIRE_SYMADDR(symaddrs->FD_Sysfd_offset, kInvalidFD);
 
   if (conn_intf.type == symaddrs->internal_syscallConn) {
+    // TODO(ddelnano): The 4.14 verifier has stricter bounds checking limits when reading memory
+    // offsets. Without this check, the verifier rejects the program. This can be removed when 4.14
+    // kernel support is dropped.
+    if (symaddrs->syscallConn_conn_offset < 0 || symaddrs->syscallConn_conn_offset > 1024) {
+      return kInvalidFD;
+    }
     REQUIRE_SYMADDR(symaddrs->syscallConn_conn_offset, kInvalidFD);
     const int kSyscallConnConnOffset = 0;
     bpf_probe_read(&conn_intf, sizeof(conn_intf),
                    conn_intf.ptr + symaddrs->syscallConn_conn_offset);
   }
 
   if (conn_intf.type == symaddrs->tls_Conn) {
+    // TODO(ddelnano): The 4.14 verifier has stricter bounds checking limits when reading memory
+    // offsets. Without this check, the verifier rejects the program. This can be removed when 4.14
+    // kernel support is dropped.
+    if (symaddrs->tlsConn_conn_offset < 0 || symaddrs->tlsConn_conn_offset > 1024) {
+      return kInvalidFD;
+    }
```

Comment:
I arbitrarily picked 1024 for this bounds check. From the opentelemetry-go-instrumentation offset file, this offset has been 0 for Go 1.19 to 1.24 ([source](https://github.com/pixie-io/opentelemetry-go-instrumentation/blob/c819b07275877c5ece92795169578261890fc186/internal/pkg/inject/offset_results.json#L4060-L4146)).